analytics/management/commands/client_activity.py

```diff
@@ -60,22 +62,27 @@ def compute_activity(self, user_activity_objects):
 
     def handle(self, *args, **options):
         # type: (*Any, **str) -> None
-        if options['arg'] is None:
+        if options['email'] is None and options['realm'] is None:
             # Report global activity.
             self.compute_activity(UserActivity.objects.all())
         else:
-            arg = options['arg']
-            try:
-                # Report activity for a user.
-                user_profile = get_user_profile_by_email(arg)
+            if options['email'] is not None and options['realm'] is None:
+                email_arg = options['email']
+                user_profile = get_user_for_mgmt(email_arg)
                 self.compute_activity(UserActivity.objects.filter(
                     user_profile=user_profile))
-            except UserProfile.DoesNotExist:
+            if options['realm'] and options['email']:
+                email_arg = options['email']
+                realmobj = get_realm(options['realm'])
+                user_profile = get_user_for_mgmt(email_arg, realmobj)
+                self.compute_activity(UserActivity.objects.filter(
+                    user_profile=user_profile))
+            else:
                 try:
                     # Report activity for a realm.
-                    realm = get_realm(arg)
+                    realmobj = get_realm(options['realm'])
                     self.compute_activity(UserActivity.objects.filter(
-                        user_profile__realm=realm))
+                        user_profile__realm=realmobj))
                 except Realm.DoesNotExist:
```

Comment:
This exception won't be raised even if the realm name is invalid as `get_realm` function uses filter query. See this blog post for more details
http://za.github.io/2015/06/29/django-filter-query-exception/